tests/torchtune/models/llama3/test_llama3_tokenizer.py

```diff
@@ -64,119 +199,361 @@ def token_ids(self):
+            False,
+            False,
+            False,
+            False,
+            False,
+            False,
+            False,
+            True,
         ]
 
-    @pytest.fixture
-    def tokenized_messages(self, token_ids):
-        return (
-            [128000, 128006, 477, 273, 128007, 10, 10]
-            + token_ids
-            + [
-                128009,
-                128006,
-                520,
-                511,
-                446,
-                128007,
-                10,
-                10,
-                65,
-                269,
-                277,
-                686,
-                334,
-                262,
-                376,
-                110,
-                351,
-                443,
-                32,
-                45,
-                334,
-                351,
-                1955,
-                46,
-                128009,
-                128001,
-            ],
-            [
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                True,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                False,
-                True,
-            ],
-        )
-
-    def test_token_ids(self, tokenizer):
-        assert tokenizer.bos_id == 128000
-        assert tokenizer.eos_id == 128001
-        assert tokenizer.pad_id == 0
-        assert tokenizer.start_header_id == 128006
-        assert tokenizer.end_header_id == 128007
-        assert tokenizer.eom_id == 128008
-        assert tokenizer.eot_id == 128009
-        assert tokenizer.python_tag == 128255
-
-    def test_tokenizer_vocab_size(self, tokenizer):
-        assert tokenizer.base_vocab_size == 2000
-        assert tokenizer.vocab_size == 128256
+    def test_tokenize_image_and_text_messages(self, tokenizer, image_and_text_messages):
```

Comment:
nit: these are all pretty much just combos of like 3-5 different token sequences, right? Might be clearest to just define them once (like `text_1_tokens`, `header_tokens`, `tool_call_tokens`, etc) then explicitly piece them together in each test to get the expected value (and can just use e.g. `len(tokens)` to build expected masks similarly). Currently reading this unit test is needlessly painful